uri/uri.h

```diff
@@ -36,6 +40,87 @@ struct Uri {
     [[nodiscard]] bool operator==(Uri const &) const = default;
 };
 
+enum HostType { DNS_DOMAIN, IP4_ADDR, IP6_ADDR, OPAQUE, EMPTY };
+
+struct Host {
+    HostType type;
+
+    // union {
```

Comment:
?

Reply:
Those should be a union, but I don't know enough C++ to know how to do that idiomatically. Unions only work the way i expect with trivial types.

Reply:
Ah, right. Yeah, unions has a bunch of requirements regarding construction/destruction and special members. Usually `std::variant<Foo, Bar, Baz>` is used instead.